bi_sql_editor/models/bi_sql_view_field.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import re
+
+from openerp import api, fields, models
+
+
+class BiSQLViewField(models.Model):
+    _name = 'bi.sql.view.field'
+    _order = 'sequence'
+
+    _TTYPE_SELECTION = [
+        ('boolean', 'boolean'),
+        ('char', 'char'),
+        ('date', 'date'),
+        ('datetime', 'datetime'),
+        ('float', 'float'),
+        ('integer', 'integer'),
+        ('many2one', 'many2one'),
+        ('selection', 'selection'),
+    ]
+
+    _GRAPH_TYPE_SELECTION = [
+        ('col', 'Column'),
+        ('row', 'Row'),
+        ('measure', 'Measure'),
+    ]
+
+    # Mapping to guess Odoo field type, from SQL column type
+    _SQL_MAPPING = {
+        'boolean': 'boolean',
+        'bigint': 'integer',
+        'integer': 'integer',
+        'double precision': 'float',
+        'numeric': 'float',
+        'text': 'char',
+        'character varying': 'char',
+        'date': 'datetime',
+        'timestamp without time zone': 'datetime',
+    }
+
+    # Mapping to try to guess most common many2one model, based on field name
+    _MODEL_MAPPING = {
```

Comment:
"_MODEL_MAPPING" can be replaced by a method that return a dict. 
like this: 
 relation_fields =  self.env['ir.model.fields'].search([('relation', '!=', False)])
dic = {}
  for f in  relation_fields:
      dict.update({f.name: f.relation'})
return dict